hangman.rb

```diff
@@ -0,0 +1,263 @@
+              "|     |       " + "|".colorize(:light_yellow),
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |       |",
+              "|     |       " + "|".colorize(:green),
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      "+ "\\".colorize(:cyan) + "|",
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|" + "/".colorize(:blue),
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/ ",
+              "|     |       | ",
+              "|     |      " + "/ ".colorize(:magenta),
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/",
+              "|     |       |",
+              "|     |      / " + "\\ ".colorize(:light_black),
+              "|     |",
+              "| ____|___"
+]]
+
+    start
+  end
+
+  def pick_word
+    word_list = %w{cat dog egg horse donkey elephant pig monkey iguana cow goat unigoat eggplant mouse dragon phoenix}
+    random_word = word_list.sample
+    return random_word
+  end
+
+  def start
+    puts "Welcome to Hangman."
+    puts "Please type your whole word or letter guess. No numbers, punctuation, or spaces."
+    # uncomment the following line for aid in development
+    puts "Answer: #{@game_word}"
+
+    @now_playing = true
+    puts "To exit, type: quit."
+    draw_tree(0)
+   
+    draw_slots
+    draw_wrong_guesses
+
+    play
+  end
+
+  def play
+    while @now_playing
+      guess = get_guess
+      check_guess(guess)
+      update_board
+      check_win
+    end
+  end
+
+  # ---------------------------------
+  # Guess Methods
+
+  def get_guess
+    input = gets.chomp.strip.downcase
+    # allow user to exit game
+    if input == "quit" || input == "exit"
+      exit
+    else
+      guess = validate_input(input)
+    end
+    return guess
+  end
+
+  def validate_input(input)
```

Comment:
Well done on this method - I like that it will go back to `get_guess` if the original input was invalid so you've covered the scenario where a user continuously inputs bad data.